src/Ormolu/Printer/Meat/Declaration/Value.hs

```diff
@@ -854,23 +854,23 @@ p_listComp s es = brackets s body
   where
     body = located es p_body
     p_body xs = do
-      let p_parBody =
-            sep
-              (breakpoint >> txt "|" >> space)
-              p_seqBody
-          p_seqBody =
-            sitcc
-              . sep
-                commaDel
-                (located' (sitcc . p_stmt))
-          stmts = init xs
-          yield = last xs
-          lists = flattenStmts stmts
+      let (stmts, yield) =
+            case xs of
+              [] -> error $ "list comprehension unexpectedly had no expressions"
+              _ -> (init xs, last xs)
```

Comment:
Maybe add a comment to use [`unsnoc`](https://hackage.haskell.org/package/base-4.20.0.1/docs/Data-List.html#v:unsnoc) as soon as we drop GHC 9.6?